### ISSUE TITLE
Reject button should always delete quest invitation (was: Keep getting invitations for an already completed quest)

### ISSUE THREAD

**Issue by ghost** [HabitRPG/habitica#7353]
_edit by admin:_

The immediate problem described below has been fixed (the bad invitation no longer exists).

**However we should change the reject button so that it always works, even if the invitation is for a quest or party that are no longer relevant. I.e., clicking the reject button should ALWAYS completely remove the invitation.**

This issue is now for implementing that change.

---

General Info
- UUID: ffa26f90-f4cf-40b5-8b3b-aca988ab8ed2
- Browser: FF 46.0.1
- OS: win10
### Description

After a quest had been started and completed by another party member for the sake of testing for https://github.com/HabitRPG/habitrpg/issues/7332, I keep getting invitation modals each time I refresh a page. Accepting or rejecting gives an error: "No quest invitation found" and "No invitation has been sent out yet" respectively.

I haven't tried the Ask Later button yet.

Screenshot: ![sc](https://snag.gy/AbXuWG.jpg)
#### Console Errors

404 POST accept https://v3.habitica.com/api/v3/groups/e3565aa7-2917-4460-9716-c930e4fd778d/quests/accept
or
404 POST reject https://v3.habitica.com/api/v3/groups/e3565aa7-2917-4460-9716-c930e4fd778d/quests/reject

and then
304 GET api/v3/user/ https://v3.habitica.com/api/v3/user/
304 GET user https://v3.habitica.com/api/v3/tasks/user

**Comment by ghost** [comment]
@paglias I've accepted the invitation for the next quest. Previous invitations are gone. Please ping me when you want me to do something or I'll just PM you my login and password.

**Comment by paglias** [comment]
@GitHubSphinx Yeah I think it happened because we did a lot of quests very fast. Probably I should allow some flexibility and just let users delete invitations even if the quest already started

**Comment by ghost** [comment]
I didn't have a chance to test the Ask Later button. So I can't say if it
could have helped. But what do you mean by delete invitations?

**Comment by paglias** [comment]
@GitHubSphinx the ask later button won't change much, it'll just ask you later. By deleting I mean that if for some reason you have an invalid quest invitation (for a quest that already started for example) you should always be able to reject it

**Comment by Alys** [comment]
@paglias I really like that idea.

**Comment by ShulamitLH** [comment]
I'm having the same problem. We have a party of 6, three of whom are daily users and 3 occassional. We just did a quest with the 3 regulars and finished it. Ever since the upgrade, the 3 occassional users have been getting invitations to that quest, both during the quest and now that it has been completed.  The quest no longer shows for the 3 of us who have completed the quest, but still shows for the 3 who didn't participate. I have each user in a separate tab in Firefox using a sandbox. Clearing the cache didn't help. Neither does either accepting or rejecting the quest and reloading and synchronizing. The invitation just keeps coming back with each reload. 

How do we delete that quest, so that we can start another quest tomorrow with all 6 users?

**Comment by Alys** [comment]
@ShulamitLH Please tell us the User IDs for each of the accounts that have that problem. We'll need to look at them to work out what's wrong.

**Comment by ShulamitLH** [comment]
It's happening with these three users in our party. They are all occasional users rather than daily users and do not participate in most quests. 

f8cc66da-b76e-45bf-96a3-0452d4d60aa1
8e347eca-a1b7-4d85-a890-a5e38479f70a
437203f0-8198-4d04-a352-5d3af826ad49

**Comment by Alys** [comment]
@ShulamitLH I've deleted the quest invitation from those three players' accounts. If you or they have more trouble, comment again here.

**devs**: I am wondering if quest invitations persist in a player's account if the quest owner starts the quest before the player has had a chance to accept or reject the invitation. That's a guess; I haven't had time to look at the code.

**Comment by hairlessbear** [comment]
@Alys You're correct. `user.party.quest.RSVPNeeded` is [one of the variables that controls the invitation modal](https://github.com/HabitRPG/habitrpg/blob/develop/website/client/js/controllers/notificationCtrl.js#L170) (`user.party.quest.completed` is the other). When a user accepts or rejects a quest, `RSVPNeeded` is set to false. If neither happens, it stays `true`, causing the modal to keep popping up forever.

This is a new bug. When `RSVPNeeded` was added in #5298, I made sure to set `RSVPNeeded` to `false` when the quest starts by using the `cleanQuestProgress` function. Amusingly, there's now [this TODO](https://github.com/HabitRPG/habitrpg/blob/develop/website/server/models/group.js#L337) in the `startQuest` function: `TODO: is it important to run clean quest progress on non-members like we did in v2?`

Why yes, it is important, to prevent this exact issue :smile:. Should be a relatively easy fix; I'll take a stab at it sometime this weekend if no one else gets to it.
